nx/lib/nx.ex

```diff
@@ -4800,6 +4800,130 @@ defmodule Nx do
     end
   end
 
+  @doc """
+  Changes the disposition of the vectorized axes of a tensor or `Nx.Container`.
+
+  Accepts the `target_axes` keyword list where the total size must match the current total
+  size of the vectorized axes. One of the items can have the `:auto` value.
+
+  ### Options
+
+    * `:target_rank` - the non-vectorized rank for the output. Can be used to transfer some of the vectorized axes
+    to the non-vectorized shape. If not given, the target rank will be at most the input tensor rank, and can be
+    reduced depending on how many `target_axes` are given.
+
+  ### Examples
+
+      iex> t = Nx.iota({1}, vectorized_axes: [x: 2, y: 3, z: 4])
+      iex> t2 = Nx.revectorize(t, x: 12, y: :auto)
+      iex> t2.vectorized_axes
+      [x: 12, y: 2]
+      iex> t3 = Nx.revectorize(t, a: :auto)
+      iex> t3.vectorized_axes
+      [a: 24]
+
+  Also works on containers. Note that the revectorization happens on a per-entry basis.
+
+      iex> t1  = Nx.iota({1}, vectorized_axes: [x: 2, y: 3])
+      iex> t2 = Nx.iota({1}, vectorized_axes: [x: 2, y: 1])
+      iex> {r1, r2} = Nx.revectorize({t1, t2}, a: :auto)
+      iex> r1.vectorized_axes
+      [a: 6]
+      iex> r2.vectorized_axes
+      [a: 2]
+
+  This function is useful for when you need to introduce a temporary custom axis to ease calculations.
+  The example below shows how to manipulate your vectorized tensor for that objective.
+
+      iex> t = Nx.iota({2}, vectorized_axes: [x: 2, y: 2])
+      #Nx.Tensor<
+        vectorized[x: 2][y: 2]
+        s64[2]
+        [
+          [
+            [0, 1],
+            [0, 1]
+          ],
+          [
+            [0, 1],
+            [0, 1]
+          ]
+        ]
+      >
+      iex> revec = Nx.revectorize(t, temp: :auto)
```

Comment:
Shouldn't the `target_rank` be here in this example. Imagine you have a tensor of rank 4. 1 dimension is vectorized and you need a rank 2 at the end. So you need to create a vectorized tensor with final shape of two.

Then at the end we revectorize it as `Nx.revectorize(result, t.vectorized_axes)`, no?

Reply:
The idea of the example is that we can introduce new axes (which happens in the line below), and `target_rank` can be used to restore the original rank.

The evolution over the example is: 2 vec + rank 1 -> 1 vec + rank 1 -> 2 vec + rank 0 -> 2 vec + rank 1

Reply:
@polvalente I may be missing something, why isn't `Nx.revectorize(result, t.vectorized_axes)` enough to reestablish it?

Reply:
Actually, I think we should accept `target_shape` that will be the inner shape of the tensor.